src/rp2_common/pico_status_led/include/pico/status_led.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2025 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/** \file pico/status_led.h
+ *  \defgroup pico_status_led pico_status_led
+ *
+ * \brief Enables access to the on-board status led
+ *
+ * Boards usually have access to an on-board status led which is configured via the board header.
+ * This library hides the details so you can use the status led for all boards without changing your code.
+ */
+
+#ifndef _PICO_STATUS_LED_H
+#define _PICO_STATUS_LED_H
+
+#include "hardware/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// PICO_CONFIG: PICO_STATUS_LED_PIN, Change the pin used for the status led from the value configured in the board headers, type=int, group=pico_status_led
+#if !defined PICO_STATUS_LED_PIN && defined PICO_DEFAULT_LED_PIN
+#define PICO_STATUS_LED_PIN PICO_DEFAULT_LED_PIN
+#endif
+
+// PICO_CONFIG: PICO_STATUS_LED_WS2812_PIN, Change the pin used for the ws2812 status led from the value configured in the board headers, type=int, group=pico_status_led
+#if !defined PICO_STATUS_LED_WS2812_PIN && defined PICO_DEFAULT_WS2812_PIN
+#define PICO_STATUS_LED_WS2812_PIN PICO_DEFAULT_WS2812_PIN
+#endif
+
+// PICO_CONFIG: PICO_STATUS_LED_USE_WS2812_PIN, If a ws2812 status pin is configured, use this instead of a normal status led , type=int, group=pico_status_led
+#if PICO_STATUS_LED_USE_WS2812_PIN && defined PICO_STATUS_LED_WS2812_PIN
+#undef PICO_STATUS_LED_PIN
+#endif
+
+/*! \brief Initialise the status led
+ *  \ingroup pico_status_led
+ *
+ *  Initialise the status led before use.
+ *
+ * \return Returns PICO_OK if the led was initialised, otherwise an error is returned
+ */
+int pico_status_led_init(void);
+
+#if defined PICO_STATUS_LED_WS2812_PIN
+
+/*! \brief Set the ws2812 status led on or off
+ *  \ingroup pico_status_led
+ *
+ * \param led_on True to turn the led on. Pass False to turn the led off
+ */
+void pico_status_led_ws2812_set(bool led_on);
+
+/*! \brief Set the ws2812 status led pixel value when on or off
```

Comment:
"status led pixel value" -> "status led colour" ?
(Also, should WS2812 be capitalised here and elsewhere?)